File: TEO/Mav.Models/PageModels/PageViewModel.cs
using System.ComponentModel.DataAnnotations;

namespace Mav.Models.PageModels
{
    public class PageViewModel
    {
        [Required]
        public int ID { get; set; }

        [Required]
        public int TypeID { get; set; }

        [Required]
        public int OrderID { get; set; }

        public int? ParentID { get; set; }

        [Required]
        public int NavigationTypeID { get; set; }

        [Required]
        public bool Published { get; set; }

        public PageInfoViewModel PageInfo { get; set; }
    }
}

File: TEO/Mav.UI/Mav.Ui/src/app/models/admin/page.view.model.ts
import { PageInfoViewModel } from './page-info.view.model';

export class PageViewModel {

  public id: number;

  public typeID: number;

  public orderID: number;

  public parentID?: number;

  public navigationTypeID: number;

  public published: boolean;

  public pageInfo: PageInfoViewModel;

}
